plugin.js

```diff
@@ -25,6 +33,21 @@ function fastifyCookieSetCookie (reply, name, value, options, signer) {
   }
 
   const serialized = cookie.serialize(name, value, opts)
+
+  if (opts.enableWarnOnSafeLimit !== false) {
+    const limit = opts.warnOnSafeLimit || 4096
+
+    const bytes = calculateCookieSize(serialized)
+
+    if (bytes > limit) {
+      reply.server.log.warn({ name, bytes }, 'Cookie[%s] has exceeded safe size limit of 4096 bytes with current size of %d bytes', name, bytes)
+    } else if (bytes === limit) {
+      reply.server.log.warn({ name, bytes }, 'Cookie[%s] has reached safe size limit of 4096 bytes with current size of %d bytes', name, bytes)
```

Comment:
why is it relevant if we "reach" the limit? Either it is safe or it is not.

Reply:
I thought it was useful to know when you're at the limit - I can remove it if you want to - I did it in the older commits too and I am also testing for it